src/python/pants/core/goals/lint.py

```diff
@@ -243,35 +372,39 @@ def key(target: type) -> str:
 
 def _print_results(
     console: Console,
-    results: tuple[LintResults, ...],
+    results_by_tool: dict[str, list[LintResult]],
     formatter_failed: bool,
 ) -> None:
-    if results:
+    if results_by_tool:
         console.print_stderr("")
 
-    for result in results:
-        if result.skipped:
-            continue
-        elif result.exit_code == 0:
-            sigil = console.sigil_succeeded()
-            status = "succeeded"
-        else:
+    for tool_name in sorted(results_by_tool):
+        results = results_by_tool[tool_name]
+        if any(result.exit_code for result in results):
             sigil = console.sigil_failed()
             status = "failed"
-        console.print_stderr(f"{sigil} {result.linter_name} {status}.")
+        else:
+            sigil = console.sigil_succeeded()
+            status = "succeeded"
+        console.print_stderr(f"{sigil} {tool_name} {status}.")
 
     if formatter_failed:
         console.print_stderr("")
         console.print_stderr(f"(One or more formatters failed. Run `{bin_name()} fmt` to fix.)")
 
 
-def _get_error_code(results: tuple[LintResults, ...]) -> int:
+def _get_error_code(results: Sequence[LintResult]) -> int:
     for result in reversed(results):
         if result.exit_code:
             return result.exit_code
     return 0
 
 
+_LintTargetsPartitionRequest = LintTargetsRequest.PartitionRequest
+_LintFilesPartitionRequest = LintFilesRequest.PartitionRequest
+_LintSubPartition = LintRequest.SubPartition
```

Comment:
These aliases are only partially used, and it's not clear to me that they add clarity. Maybe remove them?

Reply:
The `rule` parsing engine doesn't follow `Attr.Name` nodes inside a `Get` :cry:

Reply:
Can you add a TODO pointing to a ticket about that?

Reply:
I'll add a TODO.

Reply:
A comment linking to that TODO would be good here